x/mongo/driver/topology/pool.go

```diff
@@ -60,9 +66,15 @@ type checkOutResult struct {
 type pool struct {
 	address    address.Address
 	opts       []ConnectionOption
-	conns      *resourcePool // pool for non-checked out connections
 	generation *poolGenerationMap
 	monitor    *event.PoolMonitor
+	maxSize    uint
+	minSize    uint
+
+	poolChan    chan *connection
```

Comment:
(forwarding feedback to keep the conversation in the comments)

Using a buffered channel as the connection pool makes this FIFO instead of LIFO. Doesn't match the CMAP spec and is a less desirable behavior for connection management. It prevents idle connections from being closed with `maxIdleTimeMS`.